### PR TITLE
Add challenge 74: GPT-2 Inference (Medium)

### PR DESCRIPTION
**Pull request by shxjames**
## Summary

Adds challenge 73: GPT-2 Transformer Block (Medium difficulty)

Implements a single GPT-2 124M decoder block: the user receives an input activation tensor and a packed weight buffer, and must compute the full pre-norm transformer block — LayerNorm, 12-head self-attention with combined QKV projection, residual connection, LayerNorm, two-layer FFN with GELU activation, and a second residual connection

This is the first challenge that composes multiple ML primitives (LayerNorm, matmul, softmax, GELU) into a complete architectural building block, as opposed to implementing them in isolation

## What this teaches

- **End-to-end kernel composition:** fusing or scheduling multiple dependent operations (LayerNorm → matmul → softmax → matmul → GELU → matmul) with intermediate memory management
- **Packed weight buffer indexing:** all 12 parameter tensors (~7M floats / ~28MB) are packed into a single contiguous buffer at documented offsets, mimicking how real inference engines store model weights
- **Multi-head attention mechanics:** splitting projections into heads, per-head scaled dot-product attention, and concatenating results back — the core operation of transformer inference
- **Memory vs compute tradeoffs:** the attention score matrix scales as O(seq_len²) while FFN is O(seq_len), requiring different parallelization strategies for each sub-layer

## Test plan

- 10 functional tests covering: single token, seq_len 2/3/4 edge cases, 16 and 64 power-of-2, 30 and 100 non-power-of-2, 128 and 256 realistic sizes
- Performance test: seq_len=1,024 with GPT-2 124M dimensions (d_model=768, n_heads=12, ffn_dim=3,072)
- All dimensions fixed to GPT-2 124M architecture; only seq_len varies
- Tolerance atol/rtol=1e-3 to accommodate floating-point accumulation across composed operations
- All 6 framework starters present (CUDA, PyTorch, Triton, JAX, CuTe, Mojo)
- Checklist in CLAUDE.md verified

### CODE REVIEW

**Review by kunal-mansukhani — DISMISSED** [review]

File: challenges/medium/74_gpt2_block/challenge.py [hunk 1]
```diff
@@ -0,0 +1,181 @@
...
+O_BPROJ = O_WPROJ + FFN * D
+TOTAL_WEIGHTS = O_BPROJ + D
+
+
+class Challenge(ChallengeBase):
+    def __init__(self):
+        super().__init__(
+            name="GPT-2 Transformer Block",
+            atol=1e-03,
+            rtol=1e-03,
```
Comment (kunal-mansukhani) [inline challenges/medium/74_gpt2_block/challenge.py:36]: @claude Are these acceptable tolerance levels for this challenge?

File: challenges/medium/74_gpt2_block/challenge.py [hunk 2]
```diff
@@ -0,0 +1,181 @@
...
+        # layer norm 2
+        h_norm = F.layer_norm(hidden, [D], ln2_w, ln2_b, eps=1e-5)
+
+        # ffn: linear -> gelu (tanh approx) -> linear
+        fc = h_norm @ W_fc + b_fc
+        fc = F.gelu(fc, approximate="tanh")
+        proj = fc @ W_proj + b_proj
+
+        # residual connection 2
+        output.copy_(hidden + proj)
```
Comment (kunal-mansukhani) [inline challenges/medium/74_gpt2_block/challenge.py:103]: Nit: To avoid a memcpy, you can do `torch.add(hidden, proj, out=output)`
